### PR TITLE
Use a fixed version of IBM Cloud CLI for e2e tests

### LINKED ISSUE #929: pin the ibmcloud cli version
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Amulyam24**
**What this PR does / why we need it**:
Currently, in e2e tests, the latest version of IBM Cloud CLI is being downloaded. This PR adds support to use a specific stable version of the CLI.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #929 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: scripts/ci-e2e.sh [hunk 1]
```diff
@@ -93,7 +105,7 @@ prerequisites_powervs(){
     # Assigning PowerVS variables
     export IBMPOWERVS_SSHKEY_NAME=${IBMPOWERVS_SSHKEY_NAME:-"powercloud-bot-key"}
     export IBMPOWERVS_IMAGE_NAME=${IBMPOWERVS_IMAGE_NAME:-"capibm-powervs-centos-streams8-1-24-2"}
-    export IBMPOWERVS_SERVICE_INSTANCE_ID=${BOSKOS_RESOURCE_ID:-"0f28d13a-6e33-4d86-b6d7-a9b46ff7659e"}
+    export IBMPOWERVS_SERVICE_INSTANCE_ID=${BOSKOS_RESOURCE_ID:-"d53da3bf-1f4a-42fa-9735-acf16b1a05cd"}
```
Comment (mkumatag) [inline scripts/ci-e2e.sh:108]: any specific reason for changing this id? this is stale anyways and hope we are already overriding this while calling this script.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: scripts/ci-e2e.sh [hunk 2]
```diff
@@ -38,28 +34,44 @@ ARTIFACTS="${ARTIFACTS:-${PWD}/_artifacts}"
...
 }
 
+install_ibmcloud_cli(){
+    if [ ${OS} == "Linux" ]; then
+        platform="linux-${ARCH}"
+    elif [ ${OS} == "Darwin" ]; then
+        platform="macos"
+    fi
+     
+    wget https://download.clis.cloud.ibm.com/ibm-cloud-cli/${IBMCLOUD_CLI_VERSION}/binaries/IBM_Cloud_CLI_${IBMCLOUD_CLI_VERSION}_${platform}.tgz
```
Comment (mkumatag) [inline scripts/ci-e2e.sh:67]: wget will not be present by default in most of the containers, wondering if we can use curl instead.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: scripts/ci-e2e.sh (hunk 1 above)
Comment (Amulyam24) [reply scripts/ci-e2e.sh:108]: Yes, this value will be overridden. The earlier value was pointing to a deleted service instance, hence changing it. It is a placeholder value and wouldn't make much difference.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**